### PR TITLE
Added XML documentation example code parser

### PR DESCRIPTION
**Pull request by devlead**
This PR enables DSL commands for parsing example code from project xml documentation files either one file or via globber patterm.

Example usage one file
```csharp
var exampleCodes = ParseXmlDocExampleCode("./Cake.Common.xml");
foreach(var exampleCode in exampleCodes)
{
    Information(
        "{0}\r\n{1}",
        exampleCode.Name,
        exampleCode.Code
    );
}
```

Example usage globber pattern
```csharp
var filesExampleCodes = ParseXmlDocFilesExampleCode("./Cake.*.xml");
foreach(var exampleCode in filesExampleCodes)
{
    Information(
        "{0}\r\n{1}",
        exampleCode.Name,
        exampleCode.Code
    );
}
```

### CODE REVIEW

File: src/Cake.Common.Tests/Unit/Solution/Project/XmlDoc/XmlDocExampleCodeParserTests.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+                // When
+                var result = Record.Exception(() => fixture.ParseFiles());
+
+                // Then
+                Assert.IsArgumentNullException(result, "pattern");
+            }
+
+            [Fact]
+            public void Should_Return_Four_ExampleCode()
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/Solution/Project/XmlDoc/XmlDocExampleCodeParserTests.cs]: Should be called something like `Should_Return_Correct_Number_Of_Examples` or similar.

File: src/Cake.Common.Tests/Unit/Solution/Project/XmlDoc/XmlDocExampleCodeParserTests.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                var fixture = new XmlDocExampleCodeParserFixture();
+
+                // When
+                var result = fixture.Parse();
+
+                // Then
+                Assert.Equal(2, result.Count());
+            }
+        }
+        public sealed class TheParseFilesMethod
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/Solution/Project/XmlDoc/XmlDocExampleCodeParserTests.cs]: Line break.

File: src/Cake.Common/Solution/Project/XmlDoc/XmlDocAliases.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    /// Contains functionality related to MSBuild XML document files
+    /// </summary>
+    [CakeAliasCategory("MSBuild Resource")]
+    public static class XmlDocAliases
+    {
+        /// <summary>
+        /// Parses Xml documentation example code from given path
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="xmlFilePath">Path to the file to parse.</param>
```
Comment (patriksvensson) [inline src/Cake.Common/Solution/Project/XmlDoc/XmlDocAliases.cs]: Should start with `The` for consistency.

File: src/Cake.Common/Solution/Project/XmlDoc/XmlDocAliases.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+namespace Cake.Common.Solution.Project.XmlDoc
+{
+    /// <summary>
+    /// Contains functionality related to MSBuild XML document files
+    /// </summary>
+    [CakeAliasCategory("MSBuild Resource")]
+    public static class XmlDocAliases
+    {
+        /// <summary>
+        /// Parses Xml documentation example code from given path
```
Comment (patriksvensson) [inline src/Cake.Common/Solution/Project/XmlDoc/XmlDocAliases.cs]: Period at the end of the sentence.

File: src/Cake.Common/Solution/Project/XmlDoc/XmlDocAliases.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="xmlFilePath">Path to the file to parse.</param>
+        /// <returns>Parsed Example Code</returns>
```
Comment (patriksvensson) [inline src/Cake.Common/Solution/Project/XmlDoc/XmlDocAliases.cs]: Should be sentence cased (with period on the end).

File: src/Cake.Common.Tests/Fixtures/XmlDocExampleCodeParserFixture.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            Globber.GetFiles(Pattern).Returns(new FilePath[] { "/Working/Cake.Common.xml", "/Working/Cake.UnCommon.xml"});
+        }
+
+        public IEnumerable<XmlDocExampleCode> Parse()
+        {
+            var parser = new XmlDocExampleCodeParser(FileSystem, Globber);
+            return parser.Parse(XmlFilePath);
+        }
+
+
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Fixtures/XmlDocExampleCodeParserFixture.cs]: Remove line breaks.

**Comment by devlead** [comment]
@patriksvensson adjusted according to feedback and some other issued I found.

**Pull request merged**